### PR TITLE
add: cypress

### PR DESCRIPTION
**Pull request by kavyakunder**

### CODE REVIEW

File: cypress/component/ComponentName.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe('ComponentName.cy.tsx', () => {
```
Comment (shah-harshit) [inline cypress/component/ComponentName.cy.tsx]: Can you delete this files? I don't believe it is required.

File: cypress/e2e/addItem.cy.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    cy.visit("http://localhost:3000/");
+  });
+
+  it("displays the heading", () => {
+    cy.get('[data-testid="heading"]').should("contain", "Todo-List");
+    cy.get('[data-testid="btn-add"]').should("be.disabled");
+  });
+
+  it("adds a new todo item", () => {
+    cy.get('[data-testid="input-text"]').type("    ", { force: true });
```
Comment (shah-harshit) [inline cypress/e2e/addItem.cy.ts]: I don't think we need force here or anywhere else in the entire project. Can you check?

File: cypress/e2e/spec.cy.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+describe("template spec", () => {
```
Comment (shah-harshit) [inline cypress/e2e/spec.cy.ts]: I don't think we need this file as well.

File: cypress/fixtures/example.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (shah-harshit) [inline cypress/fixtures/example.json]: Not sure if we need this file. Can you check?

File: cypress/support/commands.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
+// ***********************************************
+// This example commands.ts shows you how to
+// create various custom commands and overwrite
+// existing commands.
```
Comment (shah-harshit) [inline cypress/support/commands.ts]: Same here.

File: package.json [hunk 6]
```diff
@@ -9,7 +9,6 @@
     "@mui/material": "^5.11.8",
     "@mui/styles": "^5.11.9",
     "@react-icons/all-files": "^4.1.0",
-    "@testing-library/jest-dom": "^5.16.5",
```
Comment (shah-harshit) [inline package.json]: Why is this removed from here and added to dev dependency. Let's keep it here.

File: tsconfig.json [hunk 7]
```diff
@@ -1,5 +1,7 @@
 {
   "compilerOptions": {
+    // { "compilerOptions": { "target": "es5", "lib": ["es5", "dom"], "types": ["cypress"] }, "include": ["**/*.ts"] }
```
Comment (shah-harshit) [inline tsconfig.json]: Do we need this?

File: src/App.test.tsx [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import { fireEvent, render, screen } from "@testing-library/react";
 import App from "./App";
+import "@testing-library/jest-dom";
```
Comment (shah-harshit) [inline src/App.test.tsx]: Do we need this?

File: cypress/support/e2e.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+import "./commands";
```
Comment (shah-harshit) [inline cypress/support/e2e.ts:1]: Same here.

File: cypress/support/commands.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+/// <reference types="cypress" />
```
Comment (shah-harshit) [inline cypress/support/commands.ts:1]: Same here.

File: cypress/support/e2e.ts (hunk 9 above)
Comment (kavyakunder) [reply cypress/support/e2e.ts:1]: I had removed that file, but it gave me error. So I had to put it back.

File: cypress/support/commands.ts (hunk 10 above)
Comment (kavyakunder) [reply cypress/support/commands.ts:1]: I had removed that file, but it gave me error. So I had to put it back.

**Review by shah-harshit — APPROVED** [review]
